From 4adc4039621de79bf5ae6f7218240a0b5c607225 Mon Sep 17 00:00:00 2001
From: ryanking13 <def6488@gmail.com>
Date: Tue, 22 Apr 2025 19:11:43 +0900
Subject: [PATCH 1/1] Fix _PyLong_AsByteArray function signature

The signature of _PyLong_AsByteArray has been changed in Python 3.13.
The `with_exceptions` argument was added to the function signature.

- CPython PR: https://github.com/python/cpython/pull/114886
- orjson issue: https://github.com/ijl/orjson/issues/563

---
 include/pyo3/pyo3-ffi/src/cpython/longobject.rs | 1 +
 src/serialize/per_type/int.rs                   | 1 +
 src/serialize/per_type/uuid.rs                  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/include/pyo3/pyo3-ffi/src/cpython/longobject.rs b/include/pyo3/pyo3-ffi/src/cpython/longobject.rs
index 45acaae..fb78ef1 100644
--- a/include/pyo3/pyo3-ffi/src/cpython/longobject.rs
+++ b/include/pyo3/pyo3-ffi/src/cpython/longobject.rs
@@ -68,6 +68,7 @@ extern "C" {
         n: size_t,
         little_endian: c_int,
         is_signed: c_int,
+        with_exceptions: c_int,
     ) -> c_int;
 
     // skipped _PyLong_GCD
diff --git a/src/serialize/per_type/int.rs b/src/serialize/per_type/int.rs
index 40c63f1..4ca8ed9 100644
--- a/src/serialize/per_type/int.rs
+++ b/src/serialize/per_type/int.rs
@@ -50,6 +50,7 @@ impl Serialize for IntSerializer {
                     8,
                     1,
                     is_signed,
+                    0,
                 );
                 if unlikely!(ret == -1) {
                     ffi!(PyErr_Clear());
diff --git a/src/serialize/per_type/uuid.rs b/src/serialize/per_type/uuid.rs
index 5a758b9..3838bcd 100644
--- a/src/serialize/per_type/uuid.rs
+++ b/src/serialize/per_type/uuid.rs
@@ -31,6 +31,7 @@ impl UUID {
                     16,
                     1, // little_endian
                     0, // is_signed
+                    0,
                 )
             };
             value = u128::from_le_bytes(buffer);
-- 
2.29.2.windows.2

